### ISSUE TITLE
Scheduling of a Task for a Platform fails

### ISSUE THREAD

**Issue by sadovnikov** [spring-cloud/spring-cloud-dataflow#4369]
**Description:**

Created for a Task schedule loses the Platform information, is shown twice on the Task details page and fails to display details

**Release versions:**

Version is 2.7.1

```
{"featureInfo":{"analyticsEnabled":true,"streamsEnabled":true,"tasksEnabled":true,"schedulesEnabled":true,"monitoringDashboardType":"NONE"},"versionInfo":{"implementation":{"name":"spring-cloud-dataflow-server","version":"2.7.1"},"core":{"name":"Spring Cloud Data Flow Core","version":"2.7.1"},"dashboard":{"name":"Spring Cloud Dataflow UI","version":"3.0.1"},"shell":{"name":"Spring Cloud Data Flow Shell","version":"2.7.1","url":"https://repo1.maven.org/maven2/org/springframework/cloud/spring-cloud-dataflow-shell/2.7.1/spring-cloud-dataflow-shell-2.7.1.jar"}},"securityInfo":{"authenticationEnabled":false,"authenticated":false,"username":null,"roles":[]},"runtimeEnvironment":{"appDeployer":{"deployerImplementationVersion":"2.6.1","deployerName":"Spring Cloud Skipper Server","deployerSpiVersion":"2.6.1","javaVersion":"1.8.0_282","platformApiVersion":"","platformClientVersion":"","platformHostVersion":"","platformSpecificInfo":{"sandbox":"kubernetes","default":"kubernetes","reference-apps":"kubernetes"},"platformType":"Skipper Managed","springBootVersion":"2.3.7.RELEASE","springVersion":"5.2.12.RELEASE"},"taskLaunchers":[{"deployerImplementationVersion":"2.5.1","deployerName":"KubernetesTaskLauncher","deployerSpiVersion":"2.5.1","javaVersion":"1.8.0_282","platformApiVersion":"v1","platformClientVersion":"unknown","platformHostVersion":"unknown","platformSpecificInfo":{"namespace":"sandbox","master-url":"https://172.17.224.1:443/"},"platformType":"Kubernetes","springBootVersion":"2.3.7.RELEASE","springVersion":"5.2.12.RELEASE"},{"deployerImplementationVersion":"2.5.1","deployerName":"KubernetesTaskLauncher","deployerSpiVersion":"2.5.1","javaVersion":"1.8.0_282","platformApiVersion":"v1","platformClientVersion":"unknown","platformHostVersion":"unknown","platformSpecificInfo":{"namespace":"reference-apps","master-url":"https://172.17.224.1:443/"},"platformType":"Kubernetes","springBootVersion":"2.3.7.RELEASE","springVersion":"5.2.12.RELEASE"},{"deployerImplementationVersion":"2.5.1","deployerName":"KubernetesTaskLauncher","deployerSpiVersion":"2.5.1","javaVersion":"1.8.0_282","platformApiVersion":"v1","platformClientVersion":"unknown","platformHostVersion":"unknown","platformSpecificInfo":{"namespace":"reference-apps","master-url":"https://172.17.224.1:443/"},"platformType":"Kubernetes","springBootVersion":"2.3.7.RELEASE","springVersion":"5.2.12.RELEASE"}]},"monitoringDashboardInfo":{"url":"","refreshInterval":15,"dashboardType":"NONE","source":"default-scdf-source"},"_links":{"self":{"href":"http://localhost:8080/about"}}}
```

Installed with Bitnami Helm

```
- name: spring-cloud-dataflow
  version: 2.7.0
  repository: https://charts.bitnami.com/bitnami
```

**Steps to reproduce:**

Initial state of the Task
![image](https://user-images.githubusercontent.com/4749332/106753979-b10dd500-662c-11eb-9d35-dee0171a38fc.png)

No cronjobs in the namespace
```
❯ kubectl -n reference-apps get cronjobs.batch
No resources found in reference-apps namespace.
```

Creating a Schedule for the Task
![image](https://user-images.githubusercontent.com/4749332/106754262-0fd34e80-662d-11eb-92e6-fb79b1fa2715.png)

The schedule is created, but the Platform is not shown
```
❯ kubectl -n reference-apps get cronjobs.batch
NAME                  SCHEDULE    SUSPEND   ACTIVE   LAST SCHEDULE   AGE
github-issue-report   * * * * *   False     1        32s             45s
```
![image](https://user-images.githubusercontent.com/4749332/106754379-32656780-662d-11eb-9bd3-2951f7eb70d7.png)

The Schedule appears twice on the Task details

![image](https://user-images.githubusercontent.com/4749332/106754572-6c366e00-662d-11eb-9318-c3d5402d097f.png)

Attempts to open details result in the error message in UI and logs

```
2021-02-03 13:38:15.211  WARN 1 --- [nio-8080-exec-3] o.s.c.d.s.c.RestControllerAdvice         : Caught exception while handling a request: Schedule [github-issue-report] doesn't exist
```

**Additional info:**

Our installation uses a few platforms 

![image](https://user-images.githubusercontent.com/4749332/106755260-33e35f80-662e-11eb-917b-97f66fca15d1.png)

**Comment by sabbyanandan** [comment]
Hi, @sadovnikov. Thank you for the thorough details. There appears to be flawed at many levels; we will have a look.

**Comment by sadovnikov** [comment]
@cppwfs I'll be happy to help. Changing log levels, lookups with kubectl, testing with snapshots...

**Comment by sadovnikov** [comment]
I wonder if there are any updates on this one. Thanks in advance!

**Comment by cppwfs** [comment]
There are `reference-apps` in the runtime information you have provided.   Can you verify that in your configuration that this namespace is not repeated.   If not can you provide your configuration?

**Comment by sadovnikov** [comment]
@cppwfs , thank you very much! Indeed we refer to the same `reference-apps` namespace in `default` and `reference-apps` accounts (`task.platform.kubernetes.accounts`). We want to present every available for deployments namespace as an account. I'll try removing the `default` account, although the documentation gives an impression that it's a mandatory one

**Comment by sadovnikov** [comment]
@cppwfs, removal of the `default` account, which was referring to the `reference-apps` namespace, helped to resolve duplication of the schedules on the task details page. Destroying task does result in the removal of cronjob from the namespace.

However, 
* the created schedule does not appear in the list of schedules at all. It neither appears after the server restart;
* opening details of the schedule from the task page results in `Schedule [task02] doesn't exist` message in the logs and on the screen;
* task details page fails to show the platform of the schedule.

**Comment by cppwfs** [comment]
Hello @sadovnikov ,
I was able to reproduce the initial issue.    The basic issue is that SCDF assumes that each namespace is associated with only one platform.   So when the same namespace is associated with the default platform and another platform this problem shows up.   I've created 2 issues to resolve this issue,   #4407 and #4408.   
In the meantime the default namespace is required and so for now only associated one namespace per platform.

**Comment by sadovnikov** [comment]
> SCDF assumes that each namespace is associated with only one platform

I understood this from your previous reply and removed the default account.

> In the meantime the default namespace is required 

I'll try adding a third namespace, which won't be used for deployments, as the default one

**Comment by sadovnikov** [comment]
Hello @cppwfs , I restored the "default" platform (linked to a separate namespace). However, it did not change much. 
The problem is that SCDF loses information about the platform the schedule is for.

For the same Task I created two schedules. One for a specific platform, the second one for "default".

* none of the schedules has a platform displayed on the Task details 
![image](https://user-images.githubusercontent.com/4749332/108982723-63feab00-768e-11eb-8666-5cd3cac68232.png)
* only the schedule for the "default" platform is shown in the list of schedules, and without the platform 
![image](https://user-images.githubusercontent.com/4749332/108982892-914b5900-768e-11eb-9e5b-b316e4ebd735.png)
* attempts to open details of both schedules fail

**Comment by cppwfs** [comment]
Hello @sadovnikov,
First off thanks for reporting these issues and giving great detail, it is most helpful.

Now lets get started:
Issue 1: Only able to view one platform of schedules when we may have 1 or more schedules.    
In this case you should be able to view the the schedules by platform by selecting the filter icon on the Platform column of the schedules page as shown below.
![image](https://user-images.githubusercontent.com/695673/109081072-aac0c500-76cf-11eb-9a69-7999442920a8.png)

Issue 2: Clicking schedule detail on all but one of the platforms fails.   I opened issue https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/1671 to resolve this.
Issue 3: Schedules on Schedule page and Schedules on Task Definition detail page do not show platform.   I opened Issue https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/1671 to resolve this.

**Comment by sadovnikov** [comment]
Hello @cppwfs. Thank you very much for the hint about the filter on the Schedules page. Indeed, both schedules are there. However, and this differs on your screenshots, we do not get platforms displayed there at all. Would you have any suggestions on how to debug this one?

![image](https://user-images.githubusercontent.com/4749332/109124525-4fccb380-774b-11eb-9d36-4651a1f2b9d0.png)

![image](https://user-images.githubusercontent.com/4749332/109124658-74c12680-774b-11eb-8e3c-eaffe6c0fe53.png)

**Comment by cppwfs** [comment]
Sorry for the delayed response.    
The screen I attached was based on a PR that resolves the problem of the platform not showing up:    https://github.com/spring-cloud/spring-cloud-dataflow-ui/pull/1672.  But the filtering the page by platform should work: 
![image](https://user-images.githubusercontent.com/695673/109541755-2638c280-7a92-11eb-8246-776995bad0f8.png)

**Comment by sadovnikov** [comment]
Hello @cppwfs, do I understand correctly that spring-cloud/spring-cloud-dataflow-ui#1671 is the only ticket that holds this one open?

**Comment by cppwfs** [comment]
Hello @sadovnikov,
I was awaiting your response.   But yes, https://github.com/spring-cloud/spring-cloud-dataflow-ui/pull/1671 is pending to be worked on.
Also, https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/1672 has been resolved and will be in the 2.8 release and in next release of the 2.7 as well.   
I think we can close this issue.  If you disagree, please reopen.
Again, thanks for your help!
